src/address.js

```diff
@@ -181,6 +183,28 @@ export function addOrReplaceAddressTypeOnList(addresses, newAddress) {
   return newAddresses
 }
 
+export function addOrReplaceDeliveryAddressOnList(addresses, newAddress) {
+  if (!addresses || !newAddress || isPickupAddress(newAddress)) {
+    return addresses
+  }
+
+  const newAddresses = [...addresses]
+
+  const address = getFirstAddressForDelivery(newAddresses)
+
+  if (!address) {
+    return [...newAddresses, newAddress]
+  }
+
+  const addressIndex = findAddressIndex(newAddresses, address)
+  newAddresses[addressIndex] = {
+    ...newAddresses[addressIndex],
+    ...newAddress,
+  }
```

Comment:
Hmm, this is not a `replace` action, but a `merge` action. Is it intentional? Shouldn't the function be named `addOrMergeDeliveryAddressOnList` then?

Reply:
@Lurian You are right, it is a better name. But this would made all other addOrReplace functions weird and they may already being used.
Since almost always we have a complete address passing as argument since we manipulate these objects from checkout (and they already have all the props), its almost always a replace :D

Reply:
Yeah, the ideal would be to change the other function names as well, but this would be an undesired breaking change.

I'm good with the rephrashing of the README.md to:

``` markdown
Adds a new address if no delivery address exists on addresses and the `addressType` of `newAddress` is delivery **or** replace an existing delivery address with the current delivery address merged with the `newAddress` data.
```

Reply:
Done